### PR TITLE
Support DomaEventHandler

### PR DESCRIPTION
**Pull request by making**
Pros and Cons compared to `org.seasar.doma.jdbc.entity.EntityListener`
## Pros
- POJO
- Support multiple handlers on one entity
- Support generic handler (ex. Audit Handler on all entities)
- Entity does not need to know the name of listener class
## Cons
- Need to keep up with Doma
- [`Ugly hack`](https://github.com/domaframework/doma-spring-boot/blob/cf217df726598c080b3f24b43596b5d9fa9298ca/doma-spring-boot-autoconfigure/src/main/java/org/seasar/doma/boot/event/DomaEventEntityListener.java#L7) exists because `@Entity` with default parameters uses `NullEntityListener` and forces `EntityListenerProvider` to provide `NullEntityListener` instance.

---

[Sample project](https://github.com/domaframework/doma-spring-boot/tree/eventhandler/doma-spring-boot-samples/doma-spring-boot-sample-event-handler)

### CODE REVIEW

**Comment by making** [comment]
Comments are welcome!

**Comment by nakamura-to** [comment]
Great idea!
I think that it is more useful to make EntityListenerProvider implementation configureable with properties file.

File: doma-spring-boot-autoconfigure/src/main/java/org/seasar/doma/boot/event/DomaEventEntityListener.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package org.seasar.doma.boot.event;
+
+import org.seasar.doma.jdbc.entity.*;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.context.ApplicationEventPublisherAware;
+
+public class DomaEventEntityListener<T> extends /* ugly hack */NullEntityListener<T>
```
Comment (nakamura-to) [inline doma-spring-boot-autoconfigure/src/main/java/org/seasar/doma/boot/event/DomaEventEntityListener.java]: Is it necessary to extend the `NullEntityListener` class?
You can implement the `EntityListener` interface directly.

**Comment by making** [comment]
Here are examples 
https://github.com/domaframework/doma-spring-boot/blob/71d191af65ef102be4e6608a7bde4a996e592d3e/doma-spring-boot-autoconfigure/src/test/java/org/seasar/doma/boot/event/DomaEventEntityListenerTest.java#L371-L582

**Pull request merged**